mesheryctl/internal/cli/root/filter/delete.go

```diff
@@ -27,17 +29,46 @@ mesheryctl exp filter delete test-wasm
 	Args: cobra.MinimumNArgs(0),
 	RunE: func(cmd *cobra.Command, args []string) error {
 		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		client := &http.Client{}
 		if err != nil {
 			return errors.Wrap(err, "error processing config")
 		}
+		filter := ""
+		isID := false
+		if len(args) > 0 {
+			filter = args[0]
+			// It checks if filterID is present or not
+			filterID, err := utils.GetID("filter")
+			if err == nil {
+				for _, id := range filterID {
+					if strings.HasPrefix(id, filter) {
+						filter = id
+					}
+				}
+			}
+			// check if the filter argument is a valid uuid v4 string
+			isID, err = regexp.MatchString("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$", filter)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Delete the filter using the id
+		if isID {
+			err := utils.DeleteConfiguration(filter, "filter")
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError(fmt.Sprintf("failed to delete filter %s", args[0])))
+			}
+			fmt.Printf("Filter %s deleted successfully\n", args[0])
```

Comment:
We don’t use printf, do we?

@alphaX86 @nishantwrp @hexxdump

Reply:
Seems in some areas, it's been left unchecked...

Reply:
@alphaX86 will you offer a bit of guidance to align with mesheryctl coding conventions? 

Correcting this in any other areas that have slipped by makes for good Hacktoberfest issues.

Reply:
Oh sure!

Reply:
@leecalcote that's my mistake of using printf, would make the changes accordingly.